include/swift/AST/DiagnosticsSema.def

```diff
@@ -1578,7 +1578,7 @@ ERROR(member_shadows_global_function,none,
       (DeclNameRef, DescriptiveDeclKind, const ValueDecl *, DeclName))
 
 ERROR(instance_member_use_on_type,none,
-      "instance member %1 cannot be used on type %0; "
+      "%1 non-static instance member cannot be called without an instance of type %0; "
```

Comment:
Not all members are called, though. “Referenced” might be better

Reply:
made the required changes. Kindly review.

Reply:
Note that the error in question is `could_not_use_instance_member_on_type`, not this one. They are similar.

I think we should explore the addition of a new tailored message for accesses on implicit self rather than trying to make the current one fit all targeted cases. [This](https://github.com/swiftlang/swift/pull/63301#discussion_r1092789047) is what I proposed last time.

As far as I can tell, the message targets 2 cases:

```swift
struct S {
  func instanceF() {}
}

// 1.
let m = S.self
m.instanceF()

// 2.
extension S {
  static func staticF() {
    instanceF()
  }
}
```

The GitHub issue calls out case 2.

Reply:
I am unsure about how to proceed about the changes, could you please guide me a bit.

Reply:
Use the message ID to search for the place where it is emitted. Take time to read through and examine the code around the emission site(s). Try to understand what is going on: which case is being handled, how the case is identified, what information is available about the diagnosed expression, and how you could use this data to form a condition for case 2 if we have not already. 

If you prefer Xcode, adding some breakpoints and running the compiler with a Swift file containing the test case can be very helpful. Refer to [this](https://github.com/swiftlang/swift/blob/main/docs/DebuggingTheCompiler.md#dumping-the-sil-and-other-data-in-lldb) piece of documentation for LLDB advice.

If you have questions along the way, you are welcome to post them in the PR 🙂